src/geometry/planar.md

```diff
@@ -0,0 +1,347 @@
+---
+title: Finding faces of a planar graph
+tags:
+  - Translated
+e_maxx_link: facets
+---
+# Finding faces of a planar graph
+
+Consider a graph $G$ with $n$ vertices and $m$ edges, which can be drawn on a plane in such a way that two edges intersect only at a common vertex (if it exists).
+Such graphs are called **planar**. Now suppose that we are given a planar graph together with its straight-line drawing, which means that for each vertex $v$ we have a corresponding point $(x, y)$ and all edges are drawn as line segments between these points without intersection (such drawing always exists). These line segments split the plane into several regions, which are called faces. Exactly one of the faces is unbounded. This face is called **outer**, while the other faces are called **inner**.
+
+In this article we will deal with finding both inner and outer faces of a planar graph. We will assume that the graph is connected.
+
+## Some facts about planar graphs
+
+In this section we present several facts about planar graphs without proof. Readers who are interested in proofs should refer to [Graph Theory by R. Diestel](https://sites.math.washington.edu/~billey/classes/562.winter.2018/articles/GraphTheory.pdf) or some other book.
```

Comment:
In the e-maxx articles, the proofs seem very straight-forward and mostly 1-liner. Is there a reason to refer readers to a book rather than presenting them in-place? If it's to maintain some kind of "cleanliness" of this section, we can use admonitions to "spoiler" them, by the way, like [here](https://cp-algorithms.com/algebra/continued-fractions.html).

Reply:
E-maxx proofs are hand-waving based on some geometric facts that are claimed to be obvious. I think it's better to not include any proofs than to include these.

Reply:
It seems that e-maxx proofs are based on the following facts:
1. $f=1$ when the graph is acyclic;
2. If an edge lies on a cycle, it is incident to $2$ faces, and removing the edge fuses them together;
3. In a connected graph with $n \geq 3$ vertices, each edge is incident to at most $2$ faces, and each face is incident to at least $3$ edges.

I feel like working with planar graphs in an intuitive manner would require relying on these facts anyway, so maybe we can state these facts explicitly without proving (as they're natural enough), and then prove the statements in the article, kind of relying on these facts as axioms?